prow/plugins/verify-owners/verify-owners.go

```
@@ -0,0 +1,151 @@
+	}
+
+	// Clone the repo, checkout the PR.
+	r, err := gc.Clone(pre.Repo.FullName)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := r.Clean(); err != nil {
+			log.WithError(err).Error("Error cleaning up repo.")
+		}
+	}()
+	if err := r.CheckoutPullRequest(pre.Number); err != nil {
+		return err
+	}
+
+	// Check each OWNERS file.
+	for _, f := range modifiedOwnersFiles {
+		// Try to load OWNERS file.
+		simple, err := repoowners.ParseSimpleConfig(filepath.Join(r.Dir, f))
+		if err != nil {
+			wrongOwnersFiles[f] = fmt.Errorf("error occurred parsing the OWNERS file in %s: %v", f, err)
+			break
+		}
+		// Check labels against blacklist
+		if sets.NewString(simple.Config.Labels...).HasAny(labelsBlackList...) {
+			wrongOwnersFiles[f] = fmt.Errorf("OWNERS file contains blacklisted labels: %v", sets.NewString(simple.Config.Labels...).Intersection(sets.NewString(labelsBlackList...)))
+			break
+		}
+		// Check approvers isn't empty
+		if filepath.Dir(f) == "." && len(simple.Config.Approvers) == 0 {
+			wrongOwnersFiles[f] = errors.New("no approvers defined for the repo")
+			break
+		}
+	}
+	// React if we saw something.
+	if len(wrongOwnersFiles) > 0 {
+		log.Debugf("Adding %s label.", invalidOwnersLabel)
+		if err := ghc.AddLabel(org, repo, pre.Number, invalidOwnersLabel); err != nil {
+			return err
+		}
+		log.Debugf("Creating a review for %d OWNERS files.", len(wrongOwnersFiles))
+		var comments []github.DraftReviewComment
+		for errFile, errMsg := range wrongOwnersFiles {
+			resp := fmt.Sprintf("Invalid OWNERS file: %v", errMsg)
+			comments = append(comments, github.DraftReviewComment{
+				Path: errFile,
+				Body: resp,
+			})
+		}
+		err := ghc.CreateReview(org, repo, pre.Number, github.DraftReview{
+			Action:   github.Comment,
+			Comments: comments,
+		})
+		if err != nil {
+			log.WithError(err).Error("Error creating a review for OWNERS file.")
+		}
+		return fmt.Errorf("found %d invalid OWNERS files", len(wrongOwnersFiles))
```

We don't actually want to return an error in this case since this is not a plugin error related to handling the event, but rather a problem with the PR under test.
The information in the error is already properly logged above with `log.Debugf` so just returning nil (or doing nothing since the next statement does that) is sufficient here.